### PR TITLE
Support for Seeed XIAO S3 Board

### PR DESCRIPTION
**Pull request by Dylanliacc**
# Support for Seeed XIAO S3 Board

## Summary
This PR introduces support for the Seeed XIAO S3 board, adding the necessary board and pin definitions to ensure compatibility with the Meshtastic system. Below are the detailed changes and additions:

## Board Definition: 
Added a new configuration file for the Seeed XIAO S3 board, which includes specific settings for the ESP32-S3 architecture.

## Pins Definition:
 Defined GPIO pins for various functionalities, including I2C, SPI, and LoRa module connections.
Mesh Integration: Updated the mesh protocol buffer to include the new Seeed XIAO S3 hardware model, enabling seamless integration into the existing system.

## GPS Adaptation: 
(Not recommended for non professionals to use)
Included definitions for the L76K GPS module, ensuring proper functionality when used with the XIAO S3 board. Note that there is a conflict between the RESET pin of the L76K GPS module and the SPI pins, which requires proper handling during implementation.

## Screen Adaptation: 
add oled screen via exp board.

## Future Plans and Improvements
Future work includes adapting the full range of XIAO boards, including the C6, C3, and others, to ensure comprehensive support within the Meshtastic system. Additionally, plans are in place to develop more advanced expansion boards to better support the latest display versions.

## Current related links (will be updated in the near future)
Board Information: https://www.seeedstudio.com/XIAO-ESP32S3-Sense-p-5639.html
Expansion Board Infomation : https://www.seeedstudio.com/Seeeduino-XIAO-Expansion-board-p-4746.html
L76K GPS Module Information : https://www.seeedstudio.com/L76K-GNSS-Module-for-Seeed-Studio-XIAO-p-5864.html 

Best regards,

### CODE REVIEW

**Comment by Dylanliacc** [comment]
602dcb9ac17a53a58c96577fbbaf7776

File: src/mesh/generated/meshtastic/mesh.pb.h [hunk 1]
```diff
@@ -205,6 +205,7 @@ typedef enum _meshtastic_HardwareModel {
     meshtastic_HardwareModel_RPI_PICO2 = 79,
     /* M5 esp32 based MCU modules with enclosure, TFT and LORA Shields. All Variants (Basic, Core, Fire, Core2, CoreS3, Paper) https://m5stack.com/ */
     meshtastic_HardwareModel_M5STACK_CORES3 = 80,
+    meshtastic_HardwareModel_SEEED_XIAO_S3 =81,
```
Comment (fifieldt) [inline src/mesh/generated/meshtastic/mesh.pb.h:208]: This file is automatically generated. This edit should be proposed here instead: https://github.com/meshtastic/protobufs/blob/master/meshtastic/mesh.proto

Comment (Dylanliacc) [reply src/mesh/generated/meshtastic/mesh.pb.h:208]: https://github.com/meshtastic/protobufs/pull/587   just added this PR

**Comment by caveman99** [comment]
So the Wio-SX1262 is unreleased as of yet? Looks like a nice DIY platform.

File: variants/seeed_xiao_s3/variant.h [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/*
```
Comment (fifieldt) [inline variants/seeed_xiao_s3/variant.h:1]: This file needs to be run through trunk fmt

**Comment by fifieldt** [comment]
This is very exciting :)

**Comment by Dylanliacc** [comment]
> 那么 Wio-SX1262 还没有发布吗？看起来是个不错的 DIY 平台。

in few days~ before Oct.

**Comment by Dylanliacc** [comment]
> So the Wio-SX1262 is unreleased as of yet? Looks like a nice DIY platform.

in few days ~ before Oct.

**Pull request merged**